src/containers/peacemaker/PeacemakerInformationForm.js

```diff
@@ -0,0 +1,147 @@
+// @flow
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { List, Map } from 'immutable';
+import { DataProcessingUtils, Form } from 'lattice-fabricate';
+import { CardSegment, Typography } from 'lattice-ui-kit';
+import { DataUtils, LangUtils, ReduxUtils } from 'lattice-utils';
+import type { UUID } from 'lattice';
+
+import { ADD_PEACEMAKER_INFORMATION, addPeacemakerInformation, editPeacemakerInformation } from './actions';
+import { PEACEMAKER } from './reducers/constants';
+import { schema, uiSchema } from './schemas/PeacemakerInformationSchemas';
+import { prepopulateForm } from './utils';
+
+import { CrumbItem, CrumbLink, Crumbs } from '../../components/crumbs';
+import { AppTypes, PropertyTypes } from '../../core/edm/constants';
+import {
+  APP,
+  APP_PATHS,
+  APP_REDUX_CONSTANTS,
+  EDM,
+  PROPERTY_TYPE_IDS,
+  ProfileReduxConstants,
+  REQUEST_STATE,
+} from '../../core/redux/constants';
+import { selectPerson } from '../../core/redux/selectors';
+import { getPersonName } from '../../utils/people';
+import { useDispatch, useSelector } from '../app/AppProvider';
+import { FormConstants } from '../profile/src/constants';
+
+const {
+  COMMUNICATION,
+  FORM,
+  HAS,
+  PEOPLE,
+  PERSON_DETAILS,
+  SCREENED_WITH,
+} = AppTypes;
+const { NAME } = PropertyTypes;
+const { PERSON_NEIGHBOR_MAP, PROFILE } = ProfileReduxConstants;
+const { PEACEMAKER_INFORMATION_FORM } = FormConstants;
+const { processAssociationEntityData, processEntityData } = DataProcessingUtils;
+const { isPending } = ReduxUtils;
+const { getEntityKeyId, getPropertyValue } = DataUtils;
+const { isDefined } = LangUtils;
+
+type Props = {
+  personId :UUID;
+};
+
+const PeacemakerInformationForm = ({ personId } :Props) => {
+  const dispatch = useDispatch();
+
+  const personNeighborMap :Map = useSelector((store) => store.getIn([PROFILE, PERSON_NEIGHBOR_MAP], Map()));
+
+  const forms :List = personNeighborMap.get(FORM, List());
+  const personInformationForm :?Map = forms
+    .find((form :Map) => getPropertyValue(form, [NAME, 0]) === PEACEMAKER_INFORMATION_FORM);
+  const formEKID :?UUID = isDefined(personInformationForm) ? getEntityKeyId(personInformationForm) : undefined;
+
+  const prepopulatedFormData = useMemo(() => (
+    isDefined(personInformationForm) ? prepopulateForm(personInformationForm, personNeighborMap) : {}
+  ), [personInformationForm, personNeighborMap]);
+
+  const [formData, setFormData] = useState(prepopulatedFormData);
+
+  useEffect(() => {
+    setFormData(prepopulatedFormData);
+  }, [prepopulatedFormData]);
```

Comment:
ah yes, that makes sense that the default for `formData` does not need to be `prepopulatedFormData` since `formData` is updated with `prepopulatedFormData` in the `useEffect` hook. i removed that default.

totally understand the suggestion to take out the hook, but if i were to do that, then the populated data wouldn't be updated after the user edits the info (and the data is updated in the store).

as for why "prepopulate"...good question. i had this same thought process and i think i named subsequent similar util functions "populate" instead. the only reason i can come up for why "prepopulate" is that you're populating the form before editing/submitting it. that's what the word definition says at least...but it doesn't really make sense. english is so weird.